configure tests seemed to incorrectly detect kernel features on upgrade kmod build (4.11.9-300.fc26 -> 4.13.12-200.fc26)

It seems that something about the configure script goes wrong for detecting what to set for building dattobd in Fedora 26 on upgrade from 4.11.9 to 4.13.12.

Output from make.log:
```
DKMS make.log for dattobd-0.10.3 for kernel 4.13.12-200.fc26.x86_64 (x86_64)
Sat Nov 18 21:09:45 EST 2017
make: Entering directory '/usr/src/kernels/4.13.12-200.fc26.x86_64'
  CC [M]  /var/lib/dkms/dattobd/0.10.3/build/dattobd.o
/var/lib/dkms/dattobd/0.10.3/build/dattobd.c: In function ‘dattobd_bio_endio’:
/var/lib/dkms/dattobd/0.10.3/build/dattobd.c:344:7: error: ‘struct bio’ has no member named ‘bi_error’; did you mean ‘bi_iter’?
  bio->bi_error = err;
       ^~~~~~~~
       bi_iter
/var/lib/dkms/dattobd/0.10.3/build/dattobd.c: In function ‘on_bio_read_complete’:
/var/lib/dkms/dattobd/0.10.3/build/dattobd.c:2820:35: error: ‘struct bio’ has no member named ‘bi_error’; did you mean ‘bi_iter’?
  __on_bio_read_complete(bio, bio->bi_error);
                                   ^~~~~~~~
                                   bi_iter
/var/lib/dkms/dattobd/0.10.3/build/dattobd.c: In function ‘__tracer_setup_snap’:
/var/lib/dkms/dattobd/0.10.3/build/dattobd.c:440:59: error: too few arguments to function ‘bioset_create’
 #define dattobd_bioset_create(bio_size, bvec_size, scale) bioset_create(bio_size, scale)
                                                           ^
/var/lib/dkms/dattobd/0.10.3/build/dattobd.c:3445:19: note: in expansion of macro ‘dattobd_bioset_create’
  dev->sd_bioset = dattobd_bioset_create(BIO_SET_SIZE, BIO_SET_SIZE, 0);
                   ^~~~~~~~~~~~~~~~~~~~~
In file included from ./include/linux/blkdev.h:20:0,
                 from /var/lib/dkms/dattobd/0.10.3/build/includes.h:16,
                 from /var/lib/dkms/dattobd/0.10.3/build/dattobd.c:11:
./include/linux/bio.h:390:24: note: declared here
 extern struct bio_set *bioset_create(unsigned int, unsigned int, int flags);
                        ^~~~~~~~~~~~~
make[1]: *** [scripts/Makefile.build:309: /var/lib/dkms/dattobd/0.10.3/build/dattobd.o] Error 1
make: *** [Makefile:1516: _module_/var/lib/dkms/dattobd/0.10.3/build] Error 2
make: Leaving directory '/usr/src/kernels/4.13.12-200.fc26.x86_64'
```

The running kernel at the time: 4.11.9-300.fc26
The target kernel set during the upgrade scriptlet run: 4.13.12-200.fc26

Once I rebooted into the new kernel and re-ran the postinstall scriptlet, it worked correctly. This makes me suspect something about how we detect what features we should use for building the kernel module is doing something wrong...

--- a/src/Makefile
+++ b/src/Makefile
@@ -1,6 +1,6 @@
 obj-m := dattobd.o
 
-KERNELVERSION = $(shell uname -r)
+KERNELVERSION ?= $(shell uname -r)
 KDIR := /lib/modules/$(KERNELVERSION)/build
 PWD := $(shell pwd)
 EXTRA_CFLAGS := -g

--- a/src/configure-tests/feature-tests/Makefile
+++ b/src/configure-tests/feature-tests/Makefile
@@ -1,6 +1,7 @@
 obj-m := $(OBJ)
 
-KDIR := /lib/modules/$(shell uname -r)/build
+KERNELVERSION ?= $(shell uname -r)
+KDIR := /lib/modules/$(KERNELVERSION)/build
 PWD := $(shell pwd)
 EXTRA_CFLAGS := -g -Werror
 

--- a/src/genconfig.sh
+++ b/src/genconfig.sh
@@ -1,11 +1,11 @@
 #!/bin/bash
 
-SRC_DIR=`dirname "$0"`
+SRC_DIR=$(dirname "$0")
 OUTPUT_FILE=$SRC_DIR/kernel-config.h
 FEATURE_TEST_DIR="$SRC_DIR/configure-tests/feature-tests"
 FEATURE_TEST_FILES="$FEATURE_TEST_DIR/*.c"
 SYMBOL_TESTS_FILE="$SRC_DIR/configure-tests/symbol-tests"
-KERNEL_VERSION=`uname -r`
+KERNEL_VERSION=$(uname -r)
 
 if [ ! -z "$1" ]; then
 	KERNEL_VERSION="$1"

--- a/src/genconfig.sh
+++ b/src/genconfig.sh
@@ -5,8 +5,13 @@ OUTPUT_FILE=$SRC_DIR/kernel-config.h
 FEATURE_TEST_DIR="$SRC_DIR/configure-tests/feature-tests"
 FEATURE_TEST_FILES="$FEATURE_TEST_DIR/*.c"
 SYMBOL_TESTS_FILE="$SRC_DIR/configure-tests/symbol-tests"
+KERNEL_VERSION=`uname -r`
 
-echo "generating configurations"
+if [ ! -z "$1" ]; then
+	KERNEL_VERSION="$1"
+fi
+
+echo "generating configurations for kernel-${KERNEL_VERSION}"
 
 rm -f $OUTPUT_FILE
 
@@ -16,7 +21,7 @@ echo "#ifndef DATTOBD_KERNEL_CONFIG_H" >> $OUTPUT_FILE
 echo "#define DATTOBD_KERNEL_CONFIG_H" >> $OUTPUT_FILE
 echo "" >> $OUTPUT_FILE
 
-make -s -C $FEATURE_TEST_DIR clean
+make -s -C $FEATURE_TEST_DIR clean KERNELVERSION=$KERNEL_VERSION
 
 for TEST_FILE in $FEATURE_TEST_FILES
 do
@@ -24,7 +29,7 @@ do
 	OBJ="$TEST.o"
 	MACRO_NAME="HAVE_$(echo ${TEST} | awk '{print toupper($0)}')"
 	echo -n "performing configure test: $MACRO_NAME - "
-	if make -C $FEATURE_TEST_DIR OBJ=$OBJ &>/dev/null ; then
+	if make -C $FEATURE_TEST_DIR OBJ=$OBJ KERNELVERSION=$KERNEL_VERSION &>/dev/null ; then
 		echo "present"
 		echo "#define $MACRO_NAME" >> $OUTPUT_FILE
 	else
@@ -32,7 +37,7 @@ do
 	fi
 done
 
-make -s -C $FEATURE_TEST_DIR clean
+make -s -C $FEATURE_TEST_DIR clean KERNELVERSION=$KERNEL_VERSION
 
 while read SYMBOL_NAME; do
 	if [ -z "$SYMBOL_NAME" ]; then
@@ -41,7 +46,7 @@ while read SYMBOL_NAME; do
 
 	echo "performing $SYMBOL_NAME lookup"
 	MACRO_NAME="$(echo ${SYMBOL_NAME} | awk '{print toupper($0)}')_ADDR"
-	SYMBOL_ADDR=$(grep " ${SYMBOL_NAME}$" "/boot/System.map-$1" | awk '{print $1}')
+	SYMBOL_ADDR=$(grep " ${SYMBOL_NAME}$" "/boot/System.map-${KERNEL_VERSION}" | awk '{print $1}')
 	if [ -z "$SYMBOL_ADDR" ]; then
 		SYMBOL_ADDR="0"
 	fi

